Lib/tempfile.py

```diff
@@ -848,10 +848,17 @@ def write(self, s):
         return rv
 
     def writelines(self, iterable):
-        file = self._file
-        rv = file.writelines(iterable)
-        self._check(file)
-        return rv
+        if self._max_size == 0 or self._rolled:
```

Comment:
I'd also add a test for `max_size=0` and a test which calls `writelines` twice where the first one spills to disk (to cover the `or self._rolled` part of this check).